metaflow/plugins/kfp/kfp_foreach_splits.py

```diff
@@ -22,15 +22,15 @@ def graph_to_task_ids(graph: FlowGraph) -> Dict[str, int]:
 
     Returns: node.name, or step_name -> task_id
     """
-    step_to_task_id: Dict[str, int] = {}
+    step_to_task_id: Dict[str, str] = {}
     steps_queue = ["start"]  # Queue to process the DAG in level order
     seen_steps = {"start"}  # Set of seen steps
     task_id = 0
     while len(steps_queue) > 0:
         current_step = steps_queue.pop(0)
         node = graph.nodes[current_step]
         task_id += 1
-        step_to_task_id[current_step] = task_id
+        step_to_task_id[current_step] = f'kfp-{task_id}'
```

Comment:
this is a workaround to:
https://github.com/zillow/metaflow/blob/33178350172a55d94f8ba5d8f32d51d64e11b274/metaflow/plugins/metadata/service.py#L58-L73

Let's sync with Metaflow devs on this assumption that `int` task_id are assigned by metadata service.

Reply:
prepending a string to the task id that you generate is a valid approach. just make sure the id you generate for the task is unique to the flow/run/step

Reply:
Thank you Ferras, this would in the end have no impact to the Applied Scientist experience.

I suggest `f'kfp{task_id}'` just to make it easy to copy (double click and copy) and one single string.

Reply:
updated